netbox_netprod_importer/devices_list.py

```diff
@@ -28,3 +31,53 @@ def parse_devices_yaml_def(devices_yaml, creds=None):
                     "Cannot connect to device %s: %s", hostname, e
                 )
     return devices
+
+
+def parse_filter_yaml_def(filter_yaml, creds=None):
+    netbox_api = NetboxAPI(**get_config()["netbox"])
+    devices = {}
+    with open(filter_yaml) as filter_yaml_str:
+        yml = yaml.safe_load(filter_yaml_str)
+        platforms_js = netbox_api.get("dcim/platforms")
+        platforms = {}
+        for platform in platforms_js["results"]:
+            if platform["napalm_driver"]:
+                if platform["napalm_driver"]:
+                    platforms[platform["id"]] = {
+                        "napalm_driver": platform["napalm_driver"],
+                        "napalm_args": platform["napalm_args"]
+                    }
+        if not platforms:
+            raise Exception("Not for one platform napalm_driver is not "
+                            "defined")
+
+        devlist = netbox_api.get("dcim/devices/", params=yml["filter"].items())
```

Comment:
Sorry, I made a typo before, replace that by `params=yaml["filter"]`, without the `.items()`.

Reply:
works in both cases, but without an iterator it feels longer ....

Reply:
I don't know if it is working correctly with a dict_items, maybe that's why it seems "faster". Internally, it is about same to iterate over a dict or a dict_items.

Reply:
I checked, it worked correctly. But you know better.